### PR TITLE
fix: ui onboarding misc improvements

### PR DESCRIPTION
**Pull request by adong**
## Context

Follow up PR of https://github.com/screwdriver-cd/ui/pull/547

## Objective

Improvements:

- Missing quick start guide link,
- remove git clone
- open links in new tabs
- Add requires: [~pr, ~commit], and remove steps
- warning redirect to duplicate existingId pipeline
- start with to replace select from a from
- create a pipeline first, then create a PR.
image

## References
image

Older PR: https://github.com/screwdriver-cd/ui/pull/547

More screenshots:
image

image

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by adong** [comment]
Rebased to latest master

File: app/components/create-pipeline/component.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+            this.router
+              .transitionTo('pipeline.events', pipeline.get('id'))
+              .then(() => {
+                const ctrl = getOwner(that).lookup('controller:pipeline.events');
+                const prLink = `<a href="${prUrl}" rel="noopener">${prUrl}</a>`;
+
+                ctrl.set('errorMessage', `PR: ${prLink}`);
+                that.set('showCreatePipeline', false);
+              })
```
Comment (pritamstyz4ever) [inline app/components/create-pipeline/component.js:71]: can we use async await here?

File: app/components/create-pipeline/component.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+      try {
+        pipeline = await this.store.createRecord('pipeline', payload).save();
+        this.router.transitionTo('pipeline', pipeline.get('id'));
+      } catch (err) {
+        let error = err.errors[0] || {};
+
+        if (error.status === 409 && typeof error.data === 'object' && error.data.existingId) {
+          const { existingId } = error.data;
+          const existingPiplineLink = `<a href="/pipelines/${existingId}" rel="noopener">take me there</a>`;
```
Comment (tkyi) [inline app/components/create-pipeline/component.js:43]: is this really what the link says? `take me there`?

File: app/components/create-pipeline/styles.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    // margin-left: 94px;
+    display: inline-block;
+    width: 680px;
+
+    &.menu-open {
+      margin-left: -220px;
+    }
+  }
+
+  // .right {
```
Comment (tkyi) [inline app/components/create-pipeline/styles.scss:53]: commented code

File: app/components/pipeline-create-form/styles.scss [hunk 4]
```diff
@@ -1,25 +1,44 @@
 & {
-  padding: 10px 25px 0;
+  // padding: 10px 25px 0;
```
Comment (tkyi) [inline app/components/pipeline-create-form/styles.scss:2]: commented code

File: app/components/pipeline-create-form/styles.scss [hunk 5]
```diff
@@ -1,25 +1,44 @@
...
   }
 }
 
 .text-input {
   box-sizing: border-box;
   margin-top: 10px;
   outline: 0;
-  height: 45px;
-  width: 50%;
+  // height: 45px;
```
Comment (tkyi) [inline app/components/pipeline-create-form/styles.scss:33]: more commented code

File: app/components/pipeline-create-form/styles.scss [hunk 6]
```diff
@@ -1,25 +1,44 @@
...
+  height: 34px;
+  width: 100%;
   border-radius: 2px;
   border: 2px solid $sd-separator;
-  font-size: 16px;
+  font-size: 14px;
   padding-left: 10px;
   color: $sd-text;
+  // color: rgba(48, 48, 48, 0.5);
```
Comment (tkyi) [inline app/components/pipeline-create-form/styles.scss:41]: here too

File: app/components/create-pipeline/component.js (hunk 2 above)
Comment (adong) [reply app/components/create-pipeline/component.js:43]: Yes, 😐 . We can gather more feedbacks, and change all together later.

Comment (adong) [reply app/components/create-pipeline/component.js:43]: Screenshot:
image

File: app/components/create-pipeline/component.js (hunk 1 above)
Comment (adong) [reply app/components/create-pipeline/component.js:71]: Yes. Let me change it.

Comment (adong) [reply app/components/create-pipeline/component.js:71]: screenshots:
image

File: app/components/pipeline-create-form/template.hbs [hunk 7]
```diff
@@ -1,70 +1,74 @@
 {{info-message message=errorMessage type="warning" icon="exclamation-triangle"}}
 
+<section>
+  <h1>Create Pipeline</h1>
+  <h4>For more information on how to get started with screwdriver, check out our <a href="https://docs.screwdriver.cd/user-guide/quickstart" target="_blank" rel="noopener">quickstart guide</a>.</h4>
```
Comment (tkyi) [inline app/components/pipeline-create-form/template.hbs:5]:
```suggestion
  <h4>For more information on how to get started with Screwdriver, check out our <a href="https://docs.screwdriver.cd/user-guide/quickstart" target="_blank" rel="noopener">quickstart guide</a>.</h4>
```

File: app/components/pipeline-create-form/template.hbs [hunk 8]
```diff
@@ -1,70 +1,74 @@
...
-<div>
-  {{input class="create-screwdriver-yaml" type="checkbox" checked=overrideScrewdriverYaml}}
-  <label class="normal-weight">Create <span class="select-screwdriver-yaml">Screwdriver.yaml {{fa-icon "question-circle"}}</span></label>
+  <div>
+    {{pipeline-rootdir hasRootDir=false updateRootDir=(action "updateRootDir")}}
+  </div>
+</section>
 
+<section>
+  <h5 class="bold">Create Screwdriver.yaml file</h5>
```
Comment (tkyi) [inline app/components/pipeline-create-form/template.hbs:32]:
```suggestion
  <h5 class="bold">Create screwdriver.yaml file</h5>
```

File: app/components/pipeline-create-form/template.hbs [hunk 9]
```diff
@@ -79,6 +83,6 @@
 <div class="tooltips">
   {{#bs-tooltip placement="right" triggerElement=".select-screwdriver-yaml" renderInPlace=true
   delayHide="1000"}}
-    What is <a href="https://docs.screwdriver.cd/user-guide/configuration/">screwdriver.yaml</a>
+    What is <a href="https://docs.screwdriver.cd/user-guide/configuration/" target="_blank" rel="noopener">screwdriver.yaml</a>
```
Comment (tkyi) [inline app/components/pipeline-create-form/template.hbs:86]:
```suggestion
    What is a <a href="https://docs.screwdriver.cd/user-guide/configuration/" target="_blank" rel="noopener">screwdriver.yaml</a>
```

File: app/components/quickstart-guide/styles.scss [hunk 10]
```diff
@@ -28,9 +53,10 @@
     background-color: #303030;
 
     // cover header
-    // top: 0;
+    top: 0;
+    padding-top: 25px;
     // does not cover header
-    top: 52px;
+    // top: 52px;
```
Comment (tkyi) [inline app/components/quickstart-guide/styles.scss]: commented code

File: app/components/quickstart-guide/styles.scss [hunk 11]
```diff
@@ -124,7 +157,8 @@
       > .quickstart-doc-button {
         display: flex;
         background-color: #303030;
-        border-color: $sd-light-blue;
+        border: 1px solid #80b6ff;
+        // border-color: $sd-light-blue;
```
Comment (tkyi) [inline app/components/quickstart-guide/styles.scss]: commented code

File: app/shuttle/service.js [hunk 12]
```diff
@@ -97,5 +97,24 @@ export default Service.extend({
...
+    const url = `/pipelines/${pipelineId}/openPr`;
+    const data = {
+      checkoutUrl,
+      files: [
+        {
+          name: 'screwdriver.yaml',
+          content: yaml
+        }
+      ],
+      title: 'onboard to screwdriver',
```
Comment (tkyi) [inline app/shuttle/service.js:113]:
```suggestion
      title: 'Onboard to Screwdriver',
```

File: app/shuttle/service.js [hunk 13]
```diff
@@ -97,5 +97,24 @@ export default Service.extend({
...
+      message: 'add screwdriver.yaml file'
```
Comment (tkyi) [inline app/shuttle/service.js:114]:
```suggestion
      message: 'Add screwdriver.yaml file'
```

File: tests/acceptance/create-page-test.js [hunk 14]
```diff
@@ -185,12 +185,37 @@ module('Acceptance | create', function(hooks) {
...
     await triggerEvent('.text-input', 'keyup');
-    await click('input.create-screwdriver-yaml');
 
     assert.dom('.select-template').hasText('template');
     assert.dom('.templates-dropdown').exists();
     assert.dom('.ace_editor').exists();
-    assert.dom('#template-validate-results').exists();
+  });
+
+  test('Create Screwdriver.yaml later', async function(assert) {
```
Comment (tkyi) [inline tests/acceptance/create-page-test.js:194]: this test title is kind of confusing for me

File: app/components/create-pipeline/component.js (hunk 2 above)
Comment (tkyi) [reply app/components/create-pipeline/component.js:43]: okay, I'll leave it up to you and Chas

File: tests/acceptance/create-page-test.js (hunk 14 above)
Comment (adong) [reply tests/acceptance/create-page-test.js:194]: Yeah. One test is covering if

- [x] Create Screwdriver.yaml later

is checked which user wont use embedded editor to add `screwdriver.yaml` , and one test covers if editor is shown and user uses embedded editor to create PR to add `screwdriver.yaml`

Comment (tkyi) [reply tests/acceptance/create-page-test.js:194]: I see

File: app/components/create-pipeline/component.js [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+          if (yaml && yaml.length) {
+            const pr = await this.shuttle.openPr(scmUrl, yaml);
+            const { prUrl } = pr.payload;
+            const that = this;
+
+            this.set('prUrl', prUrl);
+
+            try {
+              await this.router.transitionTo('pipeline.events', pipeline.get('id'));
+              const ctrl = getOwner(that).lookup('controller:pipeline.events');
```
Comment (jithine) [inline app/components/create-pipeline/component.js:65]: do you need `that` here ? isn't the context same fn ?

File: app/components/create-pipeline/component.js [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+              const prLink = `<a href="${prUrl}" rel="noopener">${prUrl}</a>`;
+
+              ctrl.set('errorMessage', `PR: ${prLink}`);
+              that.set('showCreatePipeline', false);
+            } catch (e) {
+              console.error('error', e);
```
Comment (jithine) [inline app/components/create-pipeline/component.js:71]: what can cause this error here ?  With `errorMessage` what's the color of the alert ? Is there a blue/green modal support ?

File: app/components/create-pipeline/component.js (hunk 15 above)
Comment (adong) [reply app/components/create-pipeline/component.js:65]: Yeah. Carried over from code refactoring. Need to update that.

File: app/components/create-pipeline/component.js (hunk 16 above)
Comment (adong) [reply app/components/create-pipeline/component.js:71]: Yellow. We can update later again.

Comment (adong) [reply app/components/create-pipeline/component.js:71]: Not any error I'd aware. Just `try catch` for `async await`

**Pull request merged**